scripts/zones/Northern_San_dOria/npcs/Grilau.lua

```diff
@@ -62,7 +62,7 @@ function onTrigger(player,npc)
             player:startEvent(1000); -- Start First Mission "Smash the Orcish scouts"
         elseif (player:hasKeyItem(tpz.ki.ANCIENT_SANDORIAN_BOOK)) then
             player:startEvent(1035);
-        elseif (CurrentMission == tpz.mission.id.sandoria.RANPERRE_S_FINAL_REST and player:getCharVar("MissionStatus",4) and tonumber(os.date("%j")) == player:getCharVar("Wait1DayForRanperre_date")) then
+        elseif (CurrentMission == tpz.mission.id.sandoria.RANPERRE_S_FINAL_REST and player:getCharVar("MissionStatus",4) and player:needToZone()) then
```

Comment:
Should this be checking `not player:needToZone()`? I'm not sure what these 3 CS's are, but it doesn't look like they're executing correctly and setting the `MissionStatus` var correctly. Everything up until this part looks right.

Reply:
want to avoid needToZone anyway since its global. we can use set/getLocalVar to tell is player zoned since the preceding step..